### ISSUE TITLE
sourcesStats every time reseting in db.

### ISSUE THREAD

**Issue by jigoop1** [Tikipeter/repository.tikipeter#448]
hi @Tikipeter 
i think this need to b mend in external.py

add  dbcur.execute("SELECT * FROM scr_perf")
after dbcur = source_utils.setPRAGMAS(dbcon) line and 
 if scraper_stats != []:
add  else: block
following works
```python
	def sourcesStats(self, sourceDict, sources):
		try:
			insert_list = []
			all_sources = [i[0] for i in sourceDict if not any(x in i[0] for x in ('season', 'show'))]
			working_scrapers = sorted(list(set([i['provider'] for i in sources])))
			non_working_scrapers = sorted([i for i in all_sources if not i in working_scrapers])
			dbcon = database.connect(self.providerDatabase, timeout=self.database_timeout)
			dbcur = source_utils.setPRAGMAS(dbcon)
			dbcur.execute("SELECT * FROM scr_perf")
			scraper_stats = dbcur.fetchall()
			if scraper_stats != []:
				for i in scraper_stats:
					try:
						scraper, success, fail = str(i[0]), i[1], i[2]
						if scraper in working_scrapers:
							insert_list.append((scraper, success+1, fail))
							working_scrapers.remove(scraper)
						else:
							insert_list.append((scraper, success, fail+1))
							non_working_scrapers.remove(scraper)
					except: pass
			else:
				if len(working_scrapers) > 0:
					for scraper in working_scrapers:
						insert_list.append((scraper, 1, 0))
				if len(non_working_scrapers) > 0:
					for scraper in non_working_scrapers:
						insert_list.append((scraper, 0, 1))
			dbcur.executemany("INSERT OR REPLACE INTO scr_perf VALUES (?, ?, ?)", insert_list)
			dbcon.commit()
			self.dbcon.close()
		except: pass
```

**Comment by Tikipeter** [comment]
Thanks @jigoop1 

I'll look into this. I don't know when I broke it to be honest.

**Comment by Tikipeter** [comment]
Fixed for next update (1.8.47).
Thanks again for the heads up jigoop1. Much appreciated.

**Comment by jigoop1** [comment]
Thank you for The Great addon.
end maintaining with prompt response.
